graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/exceptions/InvalidSubscriptionTypeException.kt

```diff
@@ -16,7 +16,13 @@
 
 package com.expediagroup.graphql.exceptions
 
+import com.expediagroup.graphql.generator.extensions.getSimpleName
 import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
 
-class InvalidSubscriptionTypeException(kClass: KClass<*>) :
-    GraphQLKotlinException("Schema requires all subscriptions to be public, ${kClass.simpleName} has ${kClass.visibility} visibility modifier")
+class InvalidSubscriptionTypeException(kClass: KClass<*>, kFunction: KFunction<*>? = null) :
```

Comment:
Is there any use case when we would throw this without having access to the `KFunction`? If not then lets make it not nullable.

Reply:
We do throw it before we check the response type when we only check the visibility so I thought the error message might make more sense if we didn't include the non-relevant info but technically we could log it for all errors

Reply:
Actually, no we don't have access to the KFunction in the first place we throw the exception when we can the entire class visibility so we do have to keep it nullable

https://github.com/ExpediaGroup/graphql-kotlin/blob/26a25015cdee58504205f49cf23840c270d156d7/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/types/SubscriptionBuilder.kt#L41-L43